### ISSUE TITLE
[Feature Request] Allow instance folder to be set per group

### ISSUE THREAD

**Issue by BinaryBurgr** [MultiMC/MultiMC5#3387]
System Information
-----------------------------
MultiMC version: 0.6.11-1430

Operating System: Windows 10 Pro 10.0.19041

Summary of the issue or suggestion: 
----------------------------------------------
Providing an option to set separate instance folders per group would allow for easy archival of old instances.

What should happen:
------------------------------
I'm imagining having a Groups page in settings where you can not only add and remove instance groups, but also optionally set an instance folder for each group. If, for example, you create a group called "Archive", then you can set that group to be stored on a larger mechanical drive and instances you put in it will automatically be moved, freeing up limited SSD space. Currently I'm moving the instance's folder manually and then symlinking back to the main instance folder, but not everyone knows how to do that and being able to manage it within MultiMC would be much cleaner.

To compliment this, it would be useful to have an indicator somewhere of how much space each instance is taking up. Perhaps some kind of Dashboard page at the top of instance settings? The Versions and/or Notes pages could maybe be rolled into it as well. There are other metrics (play time, etc.) that could go here too, but I don't want to get too off topic. 🙂 

Logs/Screenshots:
----------------------------
n/a

**Comment by Xetaxheb** [comment]
There is an export instance function that not only lets you archive unused instances but compresses things to take less space

**Comment by SirWindfield** [comment]
Does the export feature actually create a self-contained archive? Meaning that if I unzip it I can run the MC instance/modpack without having to use external programs? That would mean that it would have to bundle the JRE with it though.

If not, are there plans to add something like that? Would this feature be something you guys are interested in?

**Comment by Xetaxheb** [comment]
> Does the export feature actually create a self-contained archive? Meaning that if I unzip it I can run the MC instance/modpack without having to use external programs? That would mean that it would have to bundle the JRE with it though.

No you've.. completely misunderstood.
It's not like say AHK's "compile script" or something, it's a "put all files required for multimc to recreate this instance in a zip" and its dependencies are still: 
* java being downloaded and installed
* downloading and running multimc; 
* mojangs minecraft and forges minecraftforge distribution servers being online or the minecraft and forge installs being cached.

TL;DR it puts the following into a zip:  
* all folders (/minecraft/, /jarmods/, /patches/, anything else) 
* and files (.packignore, instance.cfg, mmc-pack.json, anything else in the top level INSTANCENAME directory folder) including options.txt; all /minecraft/mods/*.jar; and so on

But **not** any part of virtually loaded cached files like minecraft's jar or forge's jar or lwjgl (or other java libraries) or the JRE.... (multimc doesn't even contain the JRE so i'm not sure where you're pulling this from) 



> If not, are there plans to add something like that? Would this feature be something you guys are interested in?

That would require licensing to distribute the JRE or user steps to download it, plus put outdated versions of multimc in the wild, plus mojangs permission to distribute minecraft jars, plus forges permission to distribute forge, and mod authors permissions to distribute mods jars....   Unless you're using it for yourself only in which case it seems entirely pointless and you can just zip up a copy of multimc with cached mc/forges and a jre installer wherever you're thinking you'll back these up to.. if you're really that worried. 
TL;DR I don't see a point to a self-contained modpack runner? These archives are for (and already accomplish) restoring/creating a modpack in a certain state.


This is all a huge tangent though, this issue is about making it so grouping mods, like this: 
![image](https://user-images.githubusercontent.com/3267055/97983582-5d1ff580-1d8a-11eb-89b9-b06deb637222.png)
can also function as separate custom folders in different harddisks, so one can put unplayed modpacks on cheaper storage for archive instead of taking up faster storage.... but again I proffer that the export system already accomplishes this without all the "move/copy 100s of files from disk a to b" headache that changing a packs group would entail; as well as saving space since they're compressed.


> To compliment this, it would be useful to have an indicator somewhere of how much space each instance is taking up. Perhaps some kind of Dashboard page at the top of instance settings? 

i skimmed over this but personally i object to this since it means accessing and indexing the drive and that can be very slow and could increase multimc loading time significantly. if someone wants to know the folders size it takes just a couple clicks to open the mc folder and view properties.

**Comment by SirWindfield** [comment]
Ye, it was more for personal use then distributing it to other other people. The only argument that would hold up against the complexity would be decoupling from systems that need to be present in order to import the modpack again (like MultiMC, servers etc.). It would basically allow for a true offline usage.

I totally get your argument taht you could just zip the folder and you're good to go, no worries.

Maybe I'll just write something myself, some light instance manager that allows for that as it's more of a special use case for me personally.

On that note, is there a discord/channel that I can use to ask some questions about MultiMC? I don't want to clutter the issue tracker to be honest.

**Comment by Xetaxheb** [comment]
https://discord.com/invite/Sfnyvgee

**Comment by peterix** [comment]
Denied.

**Comment by peterix** [comment]
At no point am I interested in adding this.
